gubernator.go

```diff
@@ -212,50 +310,87 @@ func (s *V1Instance) asyncRequests(ctx context.Context, req *AsyncReq) {
+	span.SetTag("request.limit", req.Req.Limit)
+	span.SetTag("request.duration", req.Req.Duration)
+	span.SetTag("peer.grpcAddress", req.Peer.Info().GRPCAddress)
+
+	funcTimer := prometheus.NewTimer(funcTimeMetric.WithLabelValues("V1Instance.asyncRequests"))
+	defer funcTimer.ObserveDuration()
+
 	resp := AsyncResp{
 		Idx: req.Idx,
 	}
 
 	for {
 		if attempts > 5 {
-			resp.Resp = &RateLimitResp{
-				Error: fmt.Sprintf("GetPeer() keeps returning peers that are not connected for '%s' - '%s'", req.Key, err),
-			}
+			logrus.
+				WithError(err).
+				WithField("key", req.Key).
+				Error("GetPeer() returned peer that is not connected")
+			err2 := errors.Wrapf(err, "GetPeer() keeps returning peers that are not connected for '%s'", req.Key)
+			resp.Resp = &RateLimitResp{Error: err2.Error()}
+			countError(err, "Peer not connected")
+			ext.LogError(span, err2)
 			break
 		}
 
 		// If we are attempting again, the owner of the this rate limit might have changed to us!
 		if attempts != 0 {
 			if req.Peer.Info().IsOwner {
-				resp.Resp, err = s.getRateLimit(req.Req)
+				getRateLimitCounter.WithLabelValues("local").Add(1)
+				resp.Resp, err = s.getRateLimit(ctx, req.Req)
 				if err != nil {
-					resp.Resp = &RateLimitResp{
-						Error: fmt.Sprintf("while applying rate limit for '%s' - '%s'", req.Key, err),
-					}
+					logrus.
+						WithError(err).
+						WithField("key", req.Key).
+						Error("Error applying rate limit")
+					err2 := errors.Wrapf(err, "Error in getRateLimit for '%s'", req.Key)
+					resp.Resp = &RateLimitResp{Error: err2.Error()}
+					ext.LogError(span, err2)
 				}
 				break
 			}
 		}
 
 		// Make an RPC call to the peer that owns this rate limit
+		getRateLimitCounter.WithLabelValues("forward").Add(1)
 		r, err := req.Peer.GetPeerRateLimit(ctx, req.Req)
 		if err != nil {
 			if IsNotReady(err) {
 				attempts++
-				req.Peer, err = s.GetPeer(req.Key)
+				asyncRequestRetriesCounter.WithLabelValues(req.Req.Name).Add(1)
+				req.Peer, err = s.GetPeer(ctx, req.Key)
 				if err != nil {
-					resp.Resp = &RateLimitResp{
-						Error: fmt.Sprintf("while finding peer that owns rate limit '%s' - '%s'", req.Key, err),
-					}
+					errPart := fmt.Sprintf("Error finding peer that owns rate limit '%s'", req.Key)
+					err2 := errors.Wrap(err, errPart)
+					logrus.
+						WithError(err).
+						WithField("key", req.Key).
+						Error(errPart)
+					countError(err, "Error in GetPeer")
+					ext.LogError(span, err2)
+					resp.Resp = &RateLimitResp{Error: err2.Error()}
 					break
 				}
 				continue
 			}
-			resp.Resp = &RateLimitResp{
-				Error: fmt.Sprintf("while fetching rate limit '%s' from peer - '%s'", req.Key, err),
-			}
+
+			errPart := fmt.Sprintf("Error while fetching rate limit '%s' from peer", req.Key)
+			err2 := errors.Wrap(err, errPart)
+			logrus.
+				WithError(err).
+				WithField("key", req.Key).
+				Error("Error fetching rate limit from peer")
+			// countError(err, "Error in GetPeerRateLimit")
```

Comment:
Should this be commented out?

Reply:
Replaced with a comment explaining why `countError()` should not be called here.